app/models/transcript.rb

```diff
@@ -16,7 +16,7 @@ class Transcript < ApplicationRecord
 
   acts_as_paranoid
 
-  belongs_to :episode
+  belongs_to :episode, touch: true
```

Comment:
i noticed in the `media_resource.rb` model, that there are also `with_deleted` and `optional` and i kind of conceptually understand them but i have trouble making the connection if this model should be similar.

`with_deleted` seems to mean, even if an episode is deleted, its association would stay along? is this similar to what `dependent: :destroy` does on the episode's `has_one :transcript`? are both options needed?

and with `optional: true` (on `media_resource` for example), it seems to imply that `episode.media_resource` is not validated for presence, but right now `episode.transcript` is validated for presence? transcripts are currently optional, correct? so i should probably add in that option?

lastly, do i need to add testing for any of these options (touch, optional, with_deleted)? i didn't see anything similar in `media_resource_test.rb` so my assumption is no, but double checking with what is preferred practice.

Reply:
Probably not a bad idea to have all of `, -> { with_deleted }, touch: true, optional: true` in here.

The "with_deleted" is just means you can still access the relation even after the episode is `deleted_at`.  Imagine a deleted episode:

```ruby
Episode.where(episode_id: transcript.episode_id).first
# nil ... by default, deleted_at episodes are excluded

Episode.with_deleted.where(episode_id: transcript.episode_id).first
# now we get the episode back!
```

For `optional`... that just means `transcript.episode` is optional.  Not the other way around - the optional is on the transcript side, not the episode side.

I don't think these need to be tested.  My view is usually: let's assume rails built-ins "just work".